### PR TITLE
Update branch checkout and remote setup

### PR DESCRIPTION
**Pull request by weshaggard**

### CODE REVIEW

File: eng/pipelines/template/steps/generate-releasenotes.yml [hunk 1]
```diff
@@ -37,13 +37,27 @@ steps:
...
-        git reset --hard origin/${{ parameters.DefaultBranch }}
-        git fetch azure-sdk-fork
-        $fullPRBranchName = "${{ parameters.PrBranchName }}_${{ repo.value.Language }}_$(ReleasePeriod)"
-        echo "##vso[task.setvariable variable=FullPrBranchName]$fullPRBranchName"
-        git checkout -b $fullPRBranchName azure-sdk-fork/$fullPRBranchName 2>&1 | Out-Null
-        $LASTEXITCODE = 0 # This ignores any error from git checkout
-        git status
+
+        $RemoteName = azure-sdk-fork
+        $PRBranchName = "${{ parameters.PrBranchName }}_${{ repo.value.Language }}_$(ReleasePeriod)"
```
Comment (chidozieononiwu) [inline eng/pipelines/template/steps/generate-releasenotes.yml:42]: Does it matter that there is a parameter with the same name?

File: eng/pipelines/version-updater.yml [hunk 2]
```diff
@@ -40,6 +40,12 @@ steps:
       -github_pat '$(azuresdk-github-pat)'
       -devops_pat '$(azuresdk-azure-sdk-devops-release-work-item-pat)'
 
+- pwsh: |
+    # Some steps, like release notes, require this remote all the time so lets always setup even if there are no changes
+    git add remote azure-sdk-fork "https://$(azuresdk-github-pat)@github.com/azure-sdk/azure-sdk.git"
```
Comment (chidozieononiwu) [inline eng/pipelines/version-updater.yml:45]:
```suggestion
    git remote add azure-sdk-fork "https://$(azuresdk-github-pat)@github.com/azure-sdk/azure-sdk.git"
```

**Review by chidozieononiwu — CHANGES_REQUESTED** [review]

File: eng/pipelines/template/steps/generate-releasenotes.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/template/steps/generate-releasenotes.yml:42]: No parameters and variables are different scopes so they can have the same name.

**Review by chidozieononiwu — APPROVED** [review]

File: eng/pipelines/template/steps/generate-releasenotes.yml [hunk 3]
```diff
@@ -37,13 +37,27 @@ steps:
...
         $ErrorActionPreference = "Continue"
-        git reset --hard origin/${{ parameters.DefaultBranch }}
-        git fetch azure-sdk-fork
-        $fullPRBranchName = "${{ parameters.PrBranchName }}_${{ repo.value.Language }}_$(ReleasePeriod)"
-        echo "##vso[task.setvariable variable=FullPrBranchName]$fullPRBranchName"
-        git checkout -b $fullPRBranchName azure-sdk-fork/$fullPRBranchName 2>&1 | Out-Null
-        $LASTEXITCODE = 0 # This ignores any error from git checkout
-        git status
+
+        $RemoteName = azure-sdk-fork
```
Comment (chidozieononiwu) [inline eng/pipelines/template/steps/generate-releasenotes.yml:41]:
```suggestion
        $RemoteName = "azure-sdk-fork"
```

**Pull request merged**